src/cairoUtilFuncGen/storage/mappingIndexAccess.ts

```diff
@@ -1,41 +1,72 @@
 import assert from 'assert';
 import {
   ASTNode,
+  BytesType,
   DataLocation,
+  Expression,
   FunctionCall,
+  generalizeType,
   getNodeType,
   IndexAccess,
   MappingType,
   PointerType,
+  SourceUnit,
+  StringType,
 } from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { printNode } from '../../utils/astPrinter';
 import { CairoType, TypeConversionContext } from '../../utils/cairoTypeSystem';
 import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createUint8TypeName } from '../../utils/nodeTemplates';
 import { typeNameFromTypeNode } from '../../utils/utils';
 import { locationIfComplexType, StringIndexedFuncGen } from '../base';
+import { DynArrayGen } from './dynArray';
 
 export class MappingIndexAccessGen extends StringIndexedFuncGen {
+  private generatedHashFunctionNumber = 0;
+
+  constructor(private dynArrayGen: DynArrayGen, ast: AST, sourceUnit: SourceUnit) {
+    super(ast, sourceUnit);
+  }
+
   gen(node: IndexAccess, nodeInSourceUnit?: ASTNode): FunctionCall {
     const base = node.vBaseExpression;
-    const index = node.vIndexExpression;
+    let index = node.vIndexExpression;
     assert(index !== undefined);
 
     const nodeType = getNodeType(node, this.ast.compilerVersion);
     const baseType = getNodeType(base, this.ast.compilerVersion);
 
     assert(baseType instanceof PointerType && baseType.to instanceof MappingType);
-    const name = this.getOrCreate(
-      CairoType.fromSol(baseType.to.keyType, this.ast),
-      CairoType.fromSol(nodeType, this.ast, TypeConversionContext.StorageAllocation),
+
+    const indexCairoType = CairoType.fromSol(baseType.to.keyType, this.ast);
+    const valueCairoType = CairoType.fromSol(
+      nodeType,
+      this.ast,
+      TypeConversionContext.StorageAllocation,
     );
 
+    if (baseType.to.keyType instanceof PointerType) {
+      assert(
+        baseType.to.keyType.to instanceof StringType || baseType.to.keyType.to instanceof BytesType,
+        `Found invalid key pointer type in mapping in ${printNode(node)}`,
+      );
+      const stringLoc = generalizeType(getNodeType(index, this.ast.compilerVersion))[1];
+      assert(stringLoc !== undefined);
+      const call = this.createStringHashFunction(node, stringLoc, indexCairoType);
+      index = call;
+    }
+
+    const name = this.getOrCreate(indexCairoType, valueCairoType);
+
     const functionStub = createCairoFunctionStub(
```

Comment:
If you add true for the argument in createCarioFunctionStrub it will split the DynArray args correctly

Reply:
This relates to your convo above with Dom